### DISCUSSION TITLE
CMAKE_CROSSCOMPILING always on when building with vcpkg?

### DISCUSSION THREAD

**Discussion by klalumiere** [microsoft/vcpkg#37980] [Q&A] [answered]
I'm surprise I could not find any old issue or discussion about the subject. Maybe I missed something, but in case I did not, here it is:

I've realized that, according [to CMake documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_CROSSCOMPILING.html), if you set `CMAKE_SYSTEM_NAME` manually, `CMAKE_CROSSCOMPILING` will automatically be `TRUE`. Also, according to [vcpkg_cmake_configure doc](https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_cmake_configure), 

> [CMAKE_SYSTEM_NAME=${VCPKG_CMAKE_SYSTEM_NAME}](https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_NAME.html). If VCPKG_CMAKE_SYSTEM_NAME is unset, defaults to "Windows".

[The code is a bit more subtle](https://github.com/microsoft/vcpkg/blob/master/ports/vcpkg-cmake/vcpkg_cmake_configure.cmake), but that's roughly true.

Put both these facts together, and it would mean that `CMAKE_CROSSCOMPILING` is always ON when building with vcpkg. This has been a problem for me recently [when building cURL](https://github.com/curl/curl/blob/cfc65fd1ee164113e4b342f2e57e36fdc07c87fd/CMake/OtherTests.cmake#L78).

Am I missing something? If not, what's the best thing to do? Learn to live with `CMAKE_CROSSCOMPILING` and make patches for e.g. cURL (or add `"-DCMAKE_CROSSCOMPILING=FALSE"` manually)? Stop defining `VCPKG_CMAKE_SYSTEM_NAME` and rely on `vcpkg_cmake_configure` actual behaviour instead than relying on the documented behaviour?

Thanks! 🤓

**Comment by dg0yt** [comment, marked as answer]
There is a recent issue and PR, with this discussion:
https://github.com/microsoft/vcpkg/pull/37734
